R/tool-docs.R

```diff
@@ -91,29 +91,131 @@ btw_tool_docs_help_page <- function(topic, package_name = "") {
     check_installed(package_name)
   }
 
+  withr::local_options(list(menu.graphics = FALSE))
+
   help_page <- inject(help(
     package = !!package_name,
     topic = !!topic,
     help_type = "text",
     try.all.packages = FALSE
   ))
 
-  pager_result <- NULL
-  pager <- function(files, header, title, delete.file) {
-    str <- readLines(files, warn = FALSE)
-    str <- gsub("_\b", "", str)
-    if (isTRUE(delete.file)) {
-      unlink(files)
-    }
-    pager_result <<- str
-    invisible()
+  resolved <- help_package_topic(help_page)
+
+  if (length(resolved$resolved) > 1) {
+    calls <- sprintf(
+      '{"topic":"%s", "package_name":"%s"}',
+      resolved$resolved,
+      resolved$package
+    )
+    calls <- paste(calls, collapse = "\n")
+    resp <- sprintf(
+      "Topic '%s' matched %d different topics. Please ask the user to choose one of the following:\n%s",
```

Comment:
My reflex here is to just raise the error and let the model react (and likely choose one to look at) rather than stopping generation in this case, as we can't be sure that the tool is being called in an interactive session in some context. For example, in the context of vitals, there's no human to pick up where the conversation left off and make a choice, so ultimately the model's "solution" would be this intermediate result rather than its answer to the question it called the tool trying to resolve.